src/rviz/default_plugin/marker_display.cpp

```diff
@@ -250,9 +250,21 @@ void MarkerDisplay::incomingMarker( const visualization_msgs::Marker::ConstPtr&
 void MarkerDisplay::failedMarker(const ros::MessageEvent<visualization_msgs::Marker>& marker_evt, tf::FilterFailureReason reason)
 {
   visualization_msgs::Marker::ConstPtr marker = marker_evt.getConstMessage();
-  std::string authority = marker_evt.getPublisherName();
-  std::string error = context_->getFrameManager()->discoverFailureReason(marker->header.frame_id, marker->header.stamp, authority, reason);
-  setMarkerStatus(MarkerID(marker->ns, marker->id), StatusProperty::Error, error);
+
+  if (marker->action == visualization_msgs::Marker::DELETE)
+  {
+    processDelete(marker);
+  }
+  else if (marker->action == 3)  // TODO: visualization_msgs::Marker::DELETEALL when message changes in a future version of ROS
+  {
+    deleteAllMarkers();
+  }
+  else
+  {
+    std::string authority = marker_evt.getPublisherName();
+    std::string error = context_->getFrameManager()->discoverFailureReason(marker->header.frame_id, marker->header.stamp, authority, reason);
+    setMarkerStatus(MarkerID(marker->ns, marker->id), StatusProperty::Error, error);
+  }
```

Comment:
Would it be better to pass the marker message to `MarkerDisplay::processMessage()` if the action is `DELETE` or `DELETEALL` (3)? That was there's no need to duplicate logic. I was imagining something like this:

```c++
if (marker->action == visualization_msgs::Marker::DELETE ||
    marker->action == 3)  // TODO: visualization_msgs::Marker::DELETEALL when message changes in a future version of ROS
{
  // Process the marker anyways since it is a delete or deleteall and doesn't need to be transformed.
  return this->processMessage(marker);
}
std::string authority = marker_evt.getPublisherName();
std::string error = context_->getFrameManager()->discoverFailureReason(marker->header.frame_id, marker->header.stamp, authority, reason);
setMarkerStatus(MarkerID(marker->ns, marker->id), StatusProperty::Error, error);
```